source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -873,6 +873,10 @@ bool ShaderGraph::removeNode(ShaderNode* node)
     {
         input->breakConnection();
     }
+    for (ShaderOutput* output : node->getOutputs())
+    {
+        output->breakConnections();
+    }
```

Comment:
Without this, the downstream node's inputs connected to this output are left with a dangling pointer to it.

Reply:
See https://github.com/AcademySoftwareFoundation/MaterialX/pull/2499/files#r2573114963